sdk/typescript/src/types/common.ts

```diff
@@ -74,6 +84,51 @@ export function normalizeSuiObjectId(
   return normalizeSuiAddress(value, forceAdd0x);
 }
 
+/**
+ * Generates transaction digest.
+ *
+ * @param data transaction data
+ * @param signatureScheme signature scheme
+ * @param signature signature as a base64 string
+ * @param publicKey public key
+ */
+export function generateTransactionDigest(
+  data: TransactionData,
+  signatureScheme: SignatureScheme,
+  signature: string,
+  publicKey: PublicKeyInitData
+): string {
+  const signatureBytes = new Base64DataBuffer(signature).getData();
+
+  let pk: PublicKey;
+  switch (signatureScheme) {
+    case 'ED25519':
+      pk = new Ed25519PublicKey(publicKey);
+      break;
+    case 'Secp256k1':
+      pk = new Secp256k1PublicKey(publicKey);
+  }
+  const publicKeyBytes = pk.toBytes();
+  const schemeByte = new Uint8Array(SIGNATURE_SCHEME_TO_FLAG[signatureScheme]);
+
+  const txSignature = new Uint8Array(
+    1 + signatureBytes.length + publicKeyBytes.length
+  );
+  txSignature.set(schemeByte);
+  txSignature.set(signatureBytes, 1);
+  txSignature.set(publicKeyBytes, 1 + signatureBytes.length);
```

Comment:
i will follow up on a PR to use the serialized version of the tx signature instead of using three fields separately